torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -990,7 +929,7 @@ def state_dict(self, *args, **kwargs):
         # is available.
         if torch.cuda.is_available():
             torch.cuda.synchronize()
-        self._lazy_init()
+        _lazy_init(self, self)
```

Comment:
will there be a follow up change that replace the first `self` with a state object from the FSDP wrapper?

Reply:
Yes, state dict refactoring will come in follow-ups. All instances of `_lazy_init()` outside of the root pre-forward will look like `_lazy_init(self, self)` for now.

Reply:
This overridden `FullyShardedDataParallel.state_dict()` does not affect composable FSDP.